### ISSUE TITLE
add indirect access for matmul in online softmax

### ISSUE THREAD

**Issue by bringlein** [torch-spyre/spyre-inference#397]
once the indirect access feature is available/merged, we should replace the use of `_indirect_matmul_mock` in the Spyre attention backend. 
https://github.com/torch-spyre/spyre-inference/blob/ed58ea677989f0e65e7aefc8e72321665558534a/spyre_inference/v1/attention/backends/spyre_attn.py#L130-L197

### Done if

- [ ] all CI tests pass
- [ ] offline end2end returns meaningful results 
- [ ] Based on the final API of indirect access on spyre, the code changes should be minimal. In the best case we would just use upstream pytorch syntax (i.e. `key[page_table[i]]`). 
- [ ] `_indirect_matmul_mock` is removed.

**Comment by tdoublep** [comment]
Done via https://github.com/torch-spyre/spyre-inference/pull/539
